index.js

```diff
@@ -378,11 +378,27 @@ function Argv (processArgs, cwd) {
   }
 
   var versionOpt = null
-  self.version = function (ver, opt, msg) {
-    versionOpt = opt || 'version'
-    usage.version(ver)
+  self.version = function (opt, msg, ver) {
+    var version = null
+    var versionMsg = null
+
+    if (arguments.length === 1) {
+      version = opt
+      versionOpt = 'version'
+      versionMsg = usage.deferY18nLookup('Show version number')
+    } else if (arguments.length === 2) {
+      versionOpt = opt
+      version = msg
+      versionMsg = usage.deferY18nLookup('Show version number')
+    } else {
```

Comment:
@nexdrew @maxrimue in this pull let's concentrate on getting this logic working with appropriate tests:

```js
.version('version', '99.9')
.version('version', 'my version description', '99.9')
.version('version', 'my version description', function () {return '99.9'})
```

Then, in a follow up pull request we can use:

https://www.npmjs.com/package/find-up

To lookup the closed `package.json`, and add the following functionality:

```js
.version()
.version('version')
```

I think this is most consistent with the API elsewhere, and would simplify the version feature _a lot_.